stock_landed_costs_purchase_auto/models/purchase_order.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    landed_cost_ids = fields.One2many(
+        comodel_name="stock.landed.cost",
+        inverse_name="purchase_id",
+        string="Landed Costs",
+    )
+    landed_cost_number = fields.Integer(compute="_compute_landed_cost_number")
+
+    def _compute_landed_cost_number(self):
+        domain = [("purchase_id", "in", self.ids)]
+        res = self.env["stock.landed.cost"].read_group(
+            domain=domain, fields=["purchase_id"], groupby=["purchase_id"]
+        )
+        landed_cost_dict = {x["purchase_id"][0]: x["purchase_id_count"] for x in res}
+        for item in self:
+            item.landed_cost_number = landed_cost_dict.get(item.id, 0)
+
+    def _prepare_landed_cost_values(self, params):
+        return {"purchase_id": self.id, "picking_ids": params.get("picking_ids", False)}
+
+    def _create_picking(self):
+        res = super()._create_picking()
+        for order in self:
+            pickings = order.picking_ids.filtered(
```

Comment:
Instead of using this technique, be practical and also avoid problems with orders with no stockable products: if no picking is created, no need to create extra landed cost, so:

- Get all the pickings linked to the purchase before calling super.
- Call super.
- Get again all the linked pickings, substract from the previous one, and if there's any, then get the first one for creating the landed cost.

And another thing: you have forgotten `_create_or_update_picking`, that should use the same technique.